pkg/engine/runtime/init/init.go

```diff
@@ -1,18 +1,48 @@
 package init
 
 import (
+	"fmt"
+	"reflect"
+
 	"kusionstack.io/kusion/pkg/engine/models"
 	"kusionstack.io/kusion/pkg/engine/runtime"
 	"kusionstack.io/kusion/pkg/engine/runtime/terraform"
+	"kusionstack.io/kusion/pkg/status"
 )
 
-func InitRuntime() map[models.Type]InitFn {
-	runtimes := map[models.Type]InitFn{
-		runtime.Kubernetes: runtime.NewKubernetesRuntime,
-		runtime.Terraform:  terraform.NewTerraformRuntime,
-	}
-	return runtimes
+var SupportRuntimes = map[models.Type]InitFn{
+	runtime.Kubernetes: runtime.NewKubernetesRuntime,
+	runtime.Terraform:  terraform.NewTerraformRuntime,
 }
 
-// InitFn init Runtime
+// InitFn runtime init func
 type InitFn func() (runtime.Runtime, error)
+
+func AppendRuntimes(resources models.Resources, runtimesMap map[models.Type]runtime.Runtime) (map[models.Type]runtime.Runtime, status.Status) {
```

Comment:
I see the runtime initialization for Preview()/Apply()/Destroy(), move from pkg cmd to pkg operation, and the input param runtimesMap always be nil. `AppendRuntimes` is not a good naming if we keep to run NewXXRuntime() once for one apply action.

Reply:
I refactored this function to `Runtimes`